docs/sections/gha_workflows.md

```diff
@@ -77,6 +76,28 @@ Docker.
   * Currently, the `--reuse` functionality of the `popper run` command 
     is not available when running in Singularity.
 
+### Vagrant
+
+Popper also supports executing workflows inside Vagrant VM's.
+It spawns a VM provisioned with Docker and with the help of
+[docker-py](https://docker-py.readthedocs.io/en/stable/index.html#)
+connects to the Docker Daemon inside the VM and hence executes actions inside
+Docker containers within the VM. To execute a workflow in Vagrant runtime:
+
+```bash
+popper run --runtime vagrant
+```
+
+#### Limitations
+
+  * Currently, concurrent execution of two or more workflows is
+    not supported in Vagrant runtime as because the docker-daemon
+    in the VM can listen only on port `2375` and hence, at a
+    time if two or more VM's run each with a docker-daemon within
+    themselves, the docker client gets confused about which daemon
+    to connect to and give weird results.
```

Comment:
we could potentially run multiple workflows at the same time, as long as `VagrantRunner` checks whether the VM is already up, and skip booting it if it is. Maybe we could open an issue to work on that separately.

For now, what we can mention here that only one workflow can be executed at the time, since `popper` assumes that there is only one VM running at any given point in time (we can skip the details).